source/dataman/datasource/AMDataSourceImageData.cpp

```diff
@@ -176,36 +186,15 @@ void AMDataSourceImageData::onDataSourceDeleted()
 	setDataSource(0);
 }
 
-MPlotRange AMDataSourceImageData::range() const
-{
-	if (updateCacheRequired_)
-		updateCachedValues();
-
-	return range_;
-}
-
 void AMDataSourceImageData::updateCachedValues() const
 {
-    AMnDIndex start = AMnDIndex(0, 0);
-    AMnDIndex end = AMnDIndex(xSize_-1, ySize_-1);
-
-    if (source_->values(start, end, data_.data())){
-
-        double rangeMinimum = data_.first();
-        double rangeMaximum = data_.first();
-
-        for (int i = 0, size = data_.size(); i < size; i++){
-
-            double newValue = data_.at(i);
-
-            if (newValue > rangeMaximum)
-                rangeMaximum = newValue;
+	AMnDIndex start = AMnDIndex(0, 0);
+	AMnDIndex end = AMnDIndex(xSize_-1, ySize_-1);
 
-            if (newValue < rangeMinimum)
-                rangeMinimum = newValue;
-        }
+	if (source_->values(start, end, cachedData_.data())){
 
-        range_ = MPlotRange(rangeMinimum, rangeMaximum);
-		updateCacheRequired_ = false;
+		AMRange dataSourceRange = AMUtility::rangeFinder(cachedData_);
+		range_ = MPlotRange(dataSourceRange.minimum(), dataSourceRange.maximum());
```

Comment:
AMRange is a class I made a long time to encapsulate a range and tries to be fairly full fledged by offering things like isNull(), isValid(), etc.
MPlotRange is a typedef of a QPointF that was introduced at the beginning of MPlot because it did a decent enough job of providing the necessary functionality.